db/db_compaction_test.cc

```diff
@@ -160,7 +160,7 @@ const SstFileMetaData* PickFileRandomly(
       auto result = rand->Uniform(file_id);
       return &(level_meta.files[result]);
     }
-    file_id -= level_meta.files.size();
+    file_id -= static_cast<unsigned int>(level_meta.files.size());
```

Comment:
I feel that uint32_t is a shorter option that would preserve the size across the platforms